### PR TITLE
Update bunch of obsolete autoconf macros

### PR DESCRIPTION
**Pull request by cniethammer**
Updated macros as follows (all replacements available in autoconf 2.60):
AC_HELP_STRING --> AS_HELP_STRING
AC_ERROR --> AC_MSG_ERROR
AC_VERBOSE --> AC_MSG_NOTICE
AC_FD_CC --> AS_MESSAGE_LOG_FD
AC_CONFIG_HEADER --> AC_CONFIG_HEADERS

Signed-off-by: Christoph Niethammer <niethammer@hlrs.de>

### CODE REVIEW

**Comment by rhc54** [comment]
EC2 failure

bot:aws:retest

**Comment by rhc54** [comment]
@cniethammer Nice job! I appreciate you identifying the required changes as that will help a great deal in updating PMIx and PRRTE.

**Comment by rhc54** [comment]
@cniethammer I'm following your guide and trying to update PMIx, yet I'm getting a lot of warnings when running autogen.pl with autoconf 2.70:
```text
configure.ac:251: warning: The macro `AC_TRY_LINK' is obsolete.
configure.ac:251: You should run autoupdate.

configure.ac:251: warning: The macro `AC_TRY_COMPILE' is obsolete.
configure.ac:251: You should run autoupdate.

configure.ac:251: warning: The macro `AC_PROG_NM' is obsolete.
configure.ac:251: You should run autoupdate.

configure.ac:251: warning: The macro `_AC_COMPUTE_INT' is obsolete.
configure.ac:251: You should run autoupdate.

configure.ac:251: warning: The macro `AC_HEADER_STDC' is obsolete.
configure.ac:251: You should run autoupdate.

configure.ac:224: warning: AC_PROG_LEX without either yywrap or noyywrap is obsolete
./lib/autoconf/programs.m4:740: _AC_PROG_LEX is expanded from...
./lib/autoconf/programs.m4:714: AC_PROG_LEX is expanded from...
```
Since we copy almost all of our m4 from OMPI, I checked and those macros are still in OMPI master. Are you seeing the same warnings? Note that autogen.pl completed successfully even with all those warnings.

**Comment by rhc54** [comment]
Take a gander at https://github.com/openpmix/openpmix/pull/2029 where I have started to deal with all those warnings. I believe we'll need to do the same on OMPI.

**Comment by cniethammer** [comment]
@rhc54 yes I see also a lot of left stuff as I did not fix all warnings: (a) some of these are not straight forward, e.g., the C99 detection part, and (b) fixes are dependent on the autoconf version. So far I only used alternatives compatible with 2.60, which I found to be the required min version in ompi's configure.ac.

**Comment by rhc54** [comment]
Yeah, I think that 2.60 is too low - this 2.70 release is going to force us to raise that minimum version. Not sure how high we have to go to find fully compatible solutions. I'm using the ones available in 2.69 for now as that has been released for many years now and seems pretty available, and I believe its changes go back a little further (something like 2.67?). We should raise this issue at the next telecon.

File: config/ompi_configure_options.m4 [hunk 1]
```diff
@@ -47,7 +47,7 @@ AC_ARG_ENABLE([mpi],
 
 AC_MSG_CHECKING([if want compile-time warnings inside of mpi.h])
 AC_ARG_ENABLE(mpi-interface-warning,
```
Comment (bwbarrett) [inline config/ompi_configure_options.m4:49]: While you're making this change, can you fix the quoting as well?  I noticed a couple of places in this PR where we're missing quoting.

**Review by bwbarrett — APPROVED** [review]

File: config/ompi_configure_options.m4 (hunk 1 above)
Comment (cniethammer) [reply config/ompi_configure_options.m4]: See added commits:
* Updated the libtool macros.
* Tried to catch all missing quotings for AC_ARG_* for ompi's config files
Here my regex trying to find (hopefully) all places:
```
git grep -P 'A[CM]_ARG_[A-Z0-9_]+\([^[\)][a-z]' configure.ac config/*.m4
git grep -P '[^[]AS_HELP_STRING'
```

**Comment by awlauria** [comment]
Checking some distros:

Ubuntu 16.04/18.04/20.4/20.10: 2.69
RHEL 7.7/8.2: 2.69
openSUSE 15.2/Tumbleweed: 2.69
Amazon Linux 1: 2.69
Amazon Linux 2: can't find data

**Comment by rhc54** [comment]
@cniethammer Did you have any luck finding a replacement for `AC_PROG_NM`? I'm striking out. I simply remo,ved it and it seems to work just fine without it.

Anyone know if we really need this?

**Comment by bwbarrett** [comment]
We do still need `$NM` defined, because we use it in the assembly configure code.  However, it looks Libtool defines it these days anyway, and it's an alias for `LT_PATH_NM`.  So I think we can just call `LT_PATH_NM` and be good to go.

**Comment by rhc54** [comment]
@jsquyres suggested on Slack `AC_PATH_PROG([NM], [nm])` - would that suffice?

**Comment by bwbarrett** [comment]
I think `LT_PROG_NM` should work.  Does it not?  It actually checks more than a simple `AC_PATH_PROG`, in that it verifies that the nm found actually speaks BSD (which we rely on).

**Comment by rhc54** [comment]
Haven't tried it - will do so and report back.

**Comment by jsquyres** [comment]
I didn't look deep enough to find `LT_PATH_NM`; if that exists, it's probably a better solution.

It looks like we `AC_REQUIRE([AC_PROG_NM])` in a few places in OMPI -- that should probably be changed to `AC_REQUIRE([LT_PATH_NM])` to make sure `$NM` is set when we need it.

**Comment by cniethammer** [comment]
@jsquyres There are no other AC_PROG_NM in OMPI as far as I saw - all I found was in 3rd-party / openmpix and prrte configure.

**Comment by jsquyres** [comment]
> @jsquyres There are no other AC_PROG_NM in OMPI as far as I saw - all I found was in 3rd-party / openmpix and prrte configure.

https://github.com/open-mpi/ompi/blob/master/config/ompi_fortran_find_ext_symbol_convention.m4#L26
https://github.com/open-mpi/ompi/blob/master/config/opal_config_asm.m4#L542
https://github.com/open-mpi/ompi/blob/master/config/opal_config_asm.m4#L703

**Comment by rhc54** [comment]
Just as an FYI: in searching for solutions to some of these obsolete warnings, I have found tons of projects that are doing the update to support v2.70. I think this is something we are really just going to have to complete.

**Comment by cniethammer** [comment]
> > @jsquyres There are no other AC_PROG_NM in OMPI as far as I saw - all I found was in 3rd-party / openmpix and prrte configure.
> 
> https://github.com/open-mpi/ompi/blob/master/config/ompi_fortran_find_ext_symbol_convention.m4#L26
> https://github.com/open-mpi/ompi/blob/master/config/opal_config_asm.m4#L542
> https://github.com/open-mpi/ompi/blob/master/config/opal_config_asm.m4#L703

That's covered by my last commit in this PR from this morning - https://github.com/open-mpi/ompi/pull/8392/commits/d52673e9601f35893ec35f09acc3c6e57588d0e5

**Comment by jsquyres** [comment]
> That's covered by my last commit in this PR from this morning - [d52673e](https://github.com/open-mpi/ompi/commit/d52673e9601f35893ec35f09acc3c6e57588d0e5)

I admittedly didn't look at your patches from this morning; I was just replying to your text.  😄 

It looks like the treematch merge made a conflict with this PR.  ☹️  Can you fix?

**Review by bwbarrett — APPROVED** [review]

**Review by rhc54 — APPROVED** [review]

**Review by jsquyres — APPROVED** [review]

**Comment by jsquyres** [comment]
I think we all agree; I'll take the liberty of merging this in.

**Pull request merged**